scylla/src/cluster/metadata.rs

```diff
@@ -171,14 +171,14 @@ impl Peer {
 ///
 /// Built from `PeerEndpoint` if its `NodeAddr` variant implies address translation possibility.
 #[derive(Debug)]
-pub struct UntranslatedPeer {
+pub struct UntranslatedPeer<'a> {
     pub(crate) host_id: Uuid,
     pub(crate) untranslated_address: SocketAddr,
-    pub(crate) datacenter: Option<String>,
-    pub(crate) rack: Option<String>,
+    pub(crate) datacenter: Option<&'a str>,
+    pub(crate) rack: Option<&'a str>,
 }
 
-impl UntranslatedPeer {
+impl UntranslatedPeer<'_> {
```

Comment:
It looks like address translation is the only use of UntranslatedPeer. Can you move it to `address_translator.rs`? I see no reason for it to be in metadata.